src/eradiate/scenes/atmosphere/_particle_layer.py

```diff
@@ -441,93 +442,91 @@ def eval_radprops(
-                            dict(
-                                standard_name="absorption_coefficient",
-                                long_name="absorption coefficient",
-                                units=symbol(sigma_a.units),
-                            ),
-                        ),
-                        "sigma_s": (
-                            "z_layer",
-                            np.atleast_1d(sigma_s.magnitude),
-                            dict(
-                                standard_name="scattering_coefficient",
-                                long_name="scattering coefficient",
-                                units=symbol(sigma_s.units),
-                            ),
-                        ),
-                    }
-                )
+            ),
+        }
 
-            return xr.Dataset(
-                data_vars=data_vars,
-                coords={
-                    "z_layer": (
+        if optional_fields:
+            sigma_a = self.eval_sigma_a(spectral_ctx=spectral_ctx)
+            sigma_s = self.eval_sigma_s(spectral_ctx=spectral_ctx)
+
+            data_vars.update(
+                {
+                    "sigma_a": (
                         "z_layer",
-                        self.z_layer.magnitude,
-                        dict(
-                            standard_name="layer_altitude",
-                            long_name="layer altitude",
-                            units=symbol(self.z_layer.units),
-                        ),
-                    ),
-                    "z_level": (
-                        "z_level",
-                        self.z_level.magnitude,
+                        np.atleast_1d(sigma_a.magnitude),
                         dict(
-                            standard_name="level_altitude",
-                            long_name="level altitude",
-                            units=symbol(self.z_level.units),
+                            standard_name="absorption_coefficient",
+                            long_name="absorption coefficient",
+                            units=symbol(sigma_a.units),
                         ),
                     ),
-                    "w": (
-                        "w",
-                        [wavelength.magnitude],
+                    "sigma_s": (
+                        "z_layer",
+                        np.atleast_1d(sigma_s.magnitude),
                         dict(
-                            standard_name="wavelength",
-                            long_name="wavelength",
-                            units=symbol(wavelength.units),
+                            standard_name="scattering_coefficient",
+                            long_name="scattering coefficient",
+                            units=symbol(sigma_s.units),
                         ),
                     ),
-                },
-            ).isel(w=0)
+                }
+            )
 
-        else:
```

Comment:
If I understand correctly, incompatible configurations are detected when the eradiate.run function is called? If I'm correct, would it be feasible to have the classes constructor to raise? Detecting this kind of issue as soon as possible seems to be a good practice IMO, but if it may be complicated as I don't think it is foreseen at all. In any case implementing this would be the scope of another Issue/PR

Reply:
Things are so because I originally to make scene element instantiation mode independent. In other words, you should be able to select the mode after creating the experiment. This, in practice, doesn't really work and the pattern is more to select a mode right after import; with that in mind, transferring lode checks to constructors makes sense.